### PR TITLE
Pulled out functionality to allow MS Teams to build their message cards

### PR DESCRIPTION
**Pull request by bamandel**
Added a small bit of additional functionalty. Any other changes to this will be done in the MS Teams reformat branch

### CODE REVIEW

File: alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/ChannelMessageParser.java [hunk 1]
```diff
@@ -47,44 +47,73 @@ public String createMessage(MessageContentGroup messageContentGroup) {
...
-        if (StringUtils.isNotBlank(footerSeparator)) {
-            messagePieces.add(footerSeparator + getLineSeparator());
+        String footer = getFooter(messageContentGroup);
+        if (StringUtils.isNotBlank(footer)) {
+            messagePieces.add(footer);
         }
         return messagePieces;
     }
 
+    public String getHeader(MessageContentGroup messageContentGroup) {
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/ChannelMessageParser.java]: Can we call this `createHeader`?

File: alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/ChannelMessageParser.java [hunk 2]
```diff
@@ -47,44 +47,73 @@ public String createMessage(MessageContentGroup messageContentGroup) {
...
+                messagePieces.add(getSectionSeparator() + getLineSeparator());
+                List<String> componentItemMessagePieces = createComponentAndCategoryMessagePieces(similarItems);
+                messagePieces.addAll(componentItemMessagePieces);
+            }
+            messagePieces.add(getLineSeparator());
+            return String.join("", messagePieces);
+        }
+    }
+
+    public String getFooter(MessageContentGroup messageContentGroup) {
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/ChannelMessageParser.java]: `createFooter`?

File: alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/ChannelMessageParser.java [hunk 3]
```diff
@@ -47,44 +47,73 @@ public String createMessage(MessageContentGroup messageContentGroup) {
...
+        return createLinkableItemString(messageContentGroup.getCommonTopic(), true) + getLineSeparator();
+    }
+
+    public String getComponentSubTopic(ProviderMessageContent messageContent) {
+        return messageContent.getSubTopic()
+                   .map(item -> createLinkableItemString(item, true) + getLineSeparator())
+                   .orElse("");
+    }
+
+    public String getComponentItems(ProviderMessageContent messageContent) {
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/ChannelMessageParser.java]: `createComponentItemString`?

Comment (bamandel) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/ChannelMessageParser.java]: createComponentItemMessage?

**Pull request merged**